romancal/flatfield/tests/test_flatfield.py

```diff
@@ -54,3 +54,38 @@ def test_flatfield_step_interface(instrument, exptype):
     assert (result.data == wfi_image.data).all()
     assert result.var_flat.shape == shape
     assert result.meta.cal_step.flat_field == 'COMPLETE'
+
+
+
+@pytest.mark.parametrize(
+    "instrument, exptype",
+    [
+        ("WFI", "WFI_IMAGE"),
+    ]
+)
+@pytest.mark.skipif(
+    os.environ.get("CI") == "true",
+    reason="Roman CRDS servers are not currently available outside the internal network"
+)
+def test_crds_temporal_match(instrument, exptype):
+    """Test that the basic inferface works for data requiring a FLAT reffile"""
+
+    wfi_image = testutil.mk_level2_image(arrays=True)
+    wfi_image.meta.instrument.name = instrument
+    wfi_image.meta.instrument.detector = 'WFI01'
+    wfi_image.meta.instrument.optical_element = 'F158'
+
+    wfi_image.meta.observation.start_time = Time('2020-01-01T11:11:11.110')
+    wfi_image.meta.observation.end_time = Time('2020-01-01T11:33:11.110')
+
+    wfi_image.meta.exposure.type = exptype
+    wfi_image_model = ImageModel(wfi_image)
+
+    step = FlatFieldStep()
+    reffile = step.get_reference_file(wfi_image_model, "flat")
+    assert ("/".join(reffile.rsplit("/", 3)[1:]) == "roman/wfi/roman_wfi_flat_0057.asdf")
```

Comment:
This is going to be brittle as time goes on and replacement references are submitted to the test server.  Maybe we ought to add an argument to `step.get_reference_file` that allows us to pin the context here?

Reply:
Yeah, for this test, I feel like we are stuck with some brittleness, given that we are showing that different dates pull different flats. I a thinking to show that explicitly with something like:
`assert ("/".join(ref_file_path.rsplit("/", 3)[1:])) != ("/".join(ref_file_path_b.rsplit("/", 3)[1:]))`
where ref_file_path_b is the ref_file_path for the second file. I that with this, I could remove the flatfilename checks. What do you think?